src/handlers/expiry/providers/DateProvider.ts

```diff
@@ -0,0 +1,139 @@
+import moment from "moment-timezone";
+import { Service } from "../../../models/injector/ServiceDecorator";
+
+/**
+ * Important: The local timezone in AWS lambda is GMT for all the regions.
+ * dateFns only uses local timezones and therefore generates different hours when running locally or deployed in AWS.
+ *
+ * new Date(string) considers the ambiguous parsed string as UTC
+ * new Date() creates a new date based on the local timezone
+ */
+@Service()
+export class DateProvider {
+  private testDate: Date = new Date();
+
+  constructor() {
+    moment.tz.setDefault("UTC");
+  }
+
+  public TestDate(): Date {
+    return this.testDate;
+  }
+
+  public SetTestDate(date: Date) {
+    this.testDate = date;
+  }
+
+//#region Private Static Functions
+  public static isSameAsEpoc(inputDate: Date): boolean {
+    return moment(inputDate).isSame(new Date(0));
+  }
+
+  /**
+   * Note: moment(undefined) === moment() === new Date()
+   * @param input
+   */
+  public static isValidDate(
+    input: string | Date | number | undefined
+  ): boolean {
+    return (
+      input !== undefined &&
+      moment(input).isValid() &&
+      moment(input).isAfter(new Date(0))
+    );
+  }
+  /**
+   * To compare whether input date is between two months of compare date. Inlcusivty parameter represents start and end of month.
+   * @param inputDate the input date.
+   * @param compareDate the date to compare with.
+   * @param inclusivity []= start and end included, ()= start and end excluded, [)= start included end excluded, (]= start excluded and end included.
+   */
+  public static IsBetweenTwoMonths(
+    inputDate: Date,
+    compareDate: Date,
+    inclusivity: "[]" | "()" | "[)" | "(]" | undefined
+  ): boolean {
+    console.log(`compareDate+2 months: ${moment(compareDate).add(2, "months").toISOString()}`);
+    return moment(inputDate).isBetween(
+      moment(compareDate),
+      moment(compareDate).add(2, "months"),
+      "days",
+      inclusivity
+    );
+  }
+
+  public static IsAfterCompareDate(
+    inputDate: Date,
+    compareDate: Date
+  ): boolean {
+    return moment(inputDate).isAfter(compareDate, "days");
+  }
+
+  public static IsBeforeTwoMonths(
+    inputDate: Date,
+    compareDate: Date
+  ): boolean {
+    return moment(inputDate).isSameOrBefore(moment(compareDate).subtract(2, "months").startOf("day"), "days");
+  }
+
+  public static GetEpoc(): Date {
+    return new Date(0);
+  }
+
+
+
+  public static addOneYear(inputDate: Date | string): Date {
```

Comment:
The reason to have separate methods is some are used to create date values e.g. `regnAnniversary` in case of `HgvTrlAnnualTestStrategy`. All comparison date object and ISOString is encapsulated in DateProvider. This adds separation of concern as a result I never need to add moment as a reference in other strategy classes.